### ISSUE TITLE
imported backups / account may be forgotten

### ISSUE THREAD

**Issue by r10s** [deltachat/deltachat-ios#1504]
a once imported backup, that works for some days or hours may be forgotten, resulting in the "welcome" screen shown again.

not sure what the reason is here, initially reported via dm, however, ~~i could meanwhile also reproduce that~~ it also happens to me once.

i've seen that only for encrypted databases yet, however, not sure if that happens also to normal ones.

as similar reports are not present for android/desktop, maybe that is related to ios, maybe getting the secret from the system does not always work or so.

@cyBerta i could debug a bit: somehow the logic seems to think, there is no configured account. the account presented to the "welcome view controller" was actually empty, and there was no other account (accountIds == 1). to be honest, i played around with adding/removing accounts first, adding the same account one time encrypted and another time unencrypted. maybe that mixed up some things - however, that was a bit before the "welcome view controller" was displayed suddenly.

**Comment by plaintextdesign** [comment]
chiming in here to confirm this issue. the original backup was *not* encrypted but it was imported via "create encrypted account" and then "import backup". since then it has been (normally) imported via "log in to your server" and then "import backup". but that did not stop this behavior.

side note: encrypted accounts should be created/accessed differently!

**Comment by r10s** [comment]
> but it was imported via "create encrypted account" and then "import backup"

this is what i meant, it is currently the only way to create an encrypted account

> side note: encrypted accounts should be created/accessed differently!

yip. the current, experimental, implementation is only a step. it is intentionally a bit hidden and flagged as "experimental" to get aware of bugs as this one and to fix them before using things on a wider user base.

might be that encrypted accounts just become the default.

**Comment by cyBerta** [comment]
thanks for debugging @r10s 

> i could debug a bit: somehow the logic seems to think, there is no configured account. the account presented to the "welcome view controller" was actually empty, and there was no other account (accountIds == 1).

Could you confirm that account with the id 1 is the empty one?

**Comment by cyBerta** [comment]
I think I managed to reproduce the issue. For some reason an empty context is selected and shown. In my case the cancel button on the upper left of the AccountSetupController was visible and clickable. After clicking on it I went back to the configured account. 

I will try to figure out how to fix that.

**Comment by r10s** [comment]
> I think I managed to reproduce the issue

can you remember how you reproduced that? i am totally failing in getting the issue again :/

**Comment by r10s** [comment]
maybe related: https://github.com/deltachat/deltachat-ios/issues/1541

**Comment by cyBerta** [comment]
The only way I can reproduce it is if I try to setup a second account and then kill the app within the process. The app shows the welcome screen because of the last unconfigured account. if you tap on cancel, you get back to your configured account though.

**Comment by cyBerta** [comment]
It might be unrelated to importing backups.
@kermoshina encountered a similar issue on Android (DC v. 1.28.3) without using backups afaik. 


```
Nx: 
Hi folks! I was logged out of DC all of a sudden, when logged in back, all my message history was gone :) all my contacts and chats and stuff...
...
link2xt: 
my guess is that database got corrupted somehow on the phone (desktop?) and DC stopped recognizing it
so it asked to "log into your server" again
Looks more like a problem on the client, not the server. Similar to this bug on iOS: [https://github.com/deltachat/deltachat-ios/issues/1504](https://github.com/deltachat/deltachat-ios/issues/1504#)
```

**Comment by r10s** [comment]
maybe related: https://github.com/deltachat/deltachat-core-rust/pull/3397

**Comment by cyBerta** [comment]
closing for now, until this issue reappears

**Comment by Simon-Laux** [comment]
I had this too today with the appstore version of deltachat

**Comment by r10s** [comment]
the appstore version do not have the recent core fixes.

**Comment by Simon-Laux** [comment]
what specific core changes could have fixed it? Also I don't think its core unless there is a serious memory bug or sth like that.

**Comment by Simon-Laux** [comment]
I just got a new clue:

<img src="https://user-images.githubusercontent.com/18725968/177018108-d5470957-3ede-4cee-9560-254bed8e4b1b.png" alt="screenshot" height="350px" />

the first account is encrypted and was missing after reentering the app from sleep or background.
but it also happened on my unencrypted second device before. I'll soon upgrade to a newer core version then we can see if the bug still exists there, but before I upgrade I try to catch a log of the issue.
@r10s showed me where to get it: Login screen -> advanced -> last entry is protocol

**Comment by Simon-Laux** [comment]
So an update of what we found out:

Bug happens when getting a notification then sometimes when the encrypted accounts are closed.
`dc_is_configured` also returns 0 on error (when the account is closed it can not read the config value), so the welcome screen thought the account was unconfigured and offered the cancel button to delete it.

In #1636 I addressed how we display this state and added a guard to not allow account deletion together with an error screen that tells the user to restart the app.
The notification code still needs to be fixed, but now at least users won't loose their account to this bug anymore.
